pkg/i2gw/providers/istio/converter.go

```diff
@@ -289,10 +289,27 @@ func (c *converter) convertGateway(gw *istioclientv1beta1.Gateway, fieldPath *fi
 	}, nil
 }
 
-func (c *converter) convertVsHTTPRoutes(virtualService metav1.ObjectMeta, istioHTTPRoutes []*istiov1beta1.HTTPRoute, allowedHostnames []string, fieldPath *field.Path) ([]*gatewayv1.HTTPRoute, field.ErrorList) {
+// convertHostnames set istio hostnames as is, without extra filters.
+// If it's not a fqdn, it would be rejected by K8S API implementation
+func (c *converter) convertHostnames(hosts []string) []gatewayv1.Hostname {
+	var resHostnames []gatewayv1.Hostname
+	for _, host := range hosts {
+		// '*' is valid in istio, but not in HTTPRoute
```

Comment:
Could you add the validation to match the gateway api requirements (not only for "*" but rather like in https://github.com/kubernetes-sigs/ingress2gateway/pull/155#discussion_r1573690864)?  

Doing the check for "*" is fine but the proper validation would've been ideal to have if possible.

Reply:
updated